src/Expr.h

```diff
@@ -365,16 +367,8 @@ class Expr : public Obj {
     // Helper function to reduce boring code runs.
     StmtPtr MergeStmts(StmtPtr s1, StmtPtr s2, StmtPtr s3 = nullptr) const;
 
-    // Access to the original expression from which this one is derived,
-    // or this one if we don't have an original.  Returns a bare pointer
-    // rather than an ExprPtr to emphasize that the access is read-only.
-    const Expr* Original() const { return original ? original->Original() : this; }
-
-    // Designate the given Expr node as the original for this one.
-    void SetOriginal(ExprPtr _orig) {
-        if ( ! original )
-            original = std::move(_orig);
-    }
+    void SetLocation(const ObjPtr& orig_obj) { SetLocation(orig_obj.get()); }
+    void SetLocation(const Obj* orig_obj) { SetLocationInfo(orig_obj->GetLocationInfo()); }
```

Comment:
If we need to keep them, maybe `SetLocationFrom` or so, but if we can move the location setting into the constructors, I think these should just. For the remaining ones the explicit version works and it's not too bad to justify the helpers on first sight.
```
other->SetLocationInfo(obj->GetLocationInfo());
```

Reply:
I agree that's a better name, if we keep the approach in the PR.